### PR TITLE
added typescript support for storybook

### PR DESCRIPTION
**Pull request by jameslnewell**
#### Changes proposed in this Pull Request

* Added Typescript support to Storybook
* Added some ProgressBar examples - no CSS changes necessary to work in isolation

#### Testing instructions

* Run the Storybook and view the `ProgressBar` stories

Fixes #

### CODE REVIEW

File: packages/components/.storybook/config.js [hunk 1]
```diff
@@ -1,2 +1,4 @@
 import { configure } from '@storybook/react';
-configure( require.context( '../src', true, /\.stories\.jsx?$/ ), module );
+import '@automattic/calypso-color-schemes/dist/calypso-color-schemes.css';
```
Comment (jsnajdr) [inline packages/components/.storybook/config.js]: Did you intend to also include the color schemes CSS? Although I think it's a good idea, it doesn't look related to TypeScript 🙂

**Comment by jameslnewell** [comment]
Closing in favour of https://github.com/Automattic/wp-calypso/pull/38675
